src/razor/src/dynamicFile/dynamicFileInfoHandler.ts

```diff
@@ -40,7 +40,7 @@ export class DynamicFileInfoHandler {
             }
         );
         this.documentManager.onChange(async (e) => {
-            if (e.kind == RazorDocumentChangeKind.csharpChanged && !e.document.isOpen) {
+            if (e.kind == RazorDocumentChangeKind.csharpChanged && !e.document.isOpen && e.changes.length > 0) {
```

Comment:
```suggestion
            // Ignore any updates without text changes. This is important for perf since sending an update to roslyn does
            // a round trip for producing nothing new and causes a lot of churn in solution updates.
            if (e.kind == RazorDocumentChangeKind.csharpChanged && !e.document.isOpen && e.changes.length > 0) {
```